src/safeposix/syscalls/fs_calls.rs

```diff
@@ -915,9 +925,55 @@ impl Cage {
         }
     }
 
-    //------------------------------------CREAT SYSCALL------------------------------------
-
+    /// ## ------------------CREAT SYSCALL------------------
+    /// ### Description
+    ///
+    /// The `creat_syscall()` is similar to `open_syscall()` with the "flags"
+    /// parameter for open_syscall set to representing create, truncate or write
+    /// only for the file. It simplifies the process of creating a new file or
+    /// truncating an existing one by combining the O_CREAT, O_TRUNC, and
+    /// O_WRONLY flags.
+    ///
+    /// ### Function Arguments
+    ///
+    /// The `creat_syscall()` receives two arguments:
+    /// * `path` - This argument points to a pathname naming the file. For
+    ///   example: "/parentdir/file1" represents a file which will be either
+    ///   opened if exists or will be created at the given path.
+    /// * `mode` - This represents the permission of the newly created file. The
+    ///   general mode used is "S_IRWXA": which represents the read, write, and
+    ///   search permissions on the new file.
+    ///
+    /// ### Returns
+    ///
+    /// Upon successful completion of this call, a file descriptor is returned
+    /// which points the file which is opened. Otherwise, errors or panics
+    /// are returned for different scenarios.
+    ///
+    /// ### Errors
+    ///
+    /// * ENFILE - no available file descriptor number could be found
+    /// * ENOENT - tried to open a file that did not exist
+    /// * EPERM - the mode bits for a file are not sane
+    /// * ENOTDIR - tried to create a file as a child of something that isn't a
+    ///   directory
+    /// * EEXIST - the given file already exists
+    /// * ENXIO - the file is of type UNIX domain socket
+    ///
+    /// ### Panics
+    ///
+    /// * If truepath.file_name() returns None or if to_str() fails, causing
+    ///   unwrap() to panic.
+    /// * If the parent inode does not exist in the inode table, causing
+    ///   unwrap() to panic.
+    /// * When there is some other issue fetching the file descriptor.
+    ///
+    /// For more detailed description of all the commands and return values, see
+    /// [creat(3p)](https://man7.org/linux/man-pages/man3/creat.3p.html)
     pub fn creat_syscall(&self, path: &str, mode: u32) -> i32 {
```

Comment:
It would be great to explain what happens internally when the open_syscall is called.
You can explain specific cases about what happens when the file doesn't exist/ or exists.
(basically, give a summary about open_syscall explaining its behaviour)

Reply:
Updated the comments with the explanation of the possible cases.